### PR TITLE
Client code coverage

### PR DESCRIPTION
**Pull request by zacanbrcom**
Purpose of this PR:
- add some leftover from previous ```fetch-priority``` branch
- improve code coverage for ```CopybookDownloadService``` and ```CopybookURI```

### CODE REVIEW

**Comment by zacanbrcom** [comment]
I'd like to add in the review list also
- @grianbrcom 
- @asatklichov

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts [hunk 1]
```diff
@@ -110,3 +116,51 @@ describe("With allowed input parameters, the list of URI that represent copybook
...
+    copybookDownloadService.downloadCopybook = jest.fn().mockReturnValue("CPY");
+
+    const spySearchInWorkspace = jest.spyOn(CopybookURI, "searchInWorkspace");
+    test("With only a local folder defined in the settings.json, the search is applied locally", async () => {
+        vscode.workspace.getConfiguration = jest.fn().mockReturnValue({
+            get: jest.fn().mockReturnValue([CPY_FOLDER_NAME]),
+        });
+
+        const uri: string = await copybookURI.resolveCopybookURI(copybookName, "PRGNAME");
+        expect(uri.includes(CPY_FOLDER_NAME)).toBe(true);
```
Comment (grianbrcom) [inline clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts:142]: You could use this: `expect(uri).toMatch(CPY_FOLDER_NAME);`

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts [hunk 2]
```diff
@@ -110,3 +116,51 @@ describe("With allowed input parameters, the list of URI that represent copybook
...
+        expect(uri).toBe("");
+        expect(spySearchInWorkspace).toBeCalledTimes(2);
+    });
+
+    test("With both local and dsn references defined in the settings.json, the search is applied on local resources" +
+        "first", async () => {
+        provideMockValueForLocalAndDSN(CPY_FOLDER_NAME, "");
+
+        const uri: string = await copybookURI.resolveCopybookURI(copybookName, "PRGNAME");
+        expect(uri === "").toBe(false);
```
Comment (grianbrcom) [inline clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts:162]: Another example: `expect(uri).not.toBe("");`

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts [hunk 3]
```diff
@@ -12,119 +12,207 @@
...
-        expect(vscode.window.showErrorMessage)
-            .toHaveBeenCalledWith("Connection to mainframe using Zowe profile zoweProfile failed.");
+});
+
+describe("Test the creation of folders that contains copybooks downloaded from MF against correct configuration in settings provided by the user", () => {
+    function setupScenario() {
+        const testFolder = path.join(__dirname, C4Z_FOLDER, COPYBOOKS_FOLDER, "profile", "dataset");
+        const copybookURIPath = path.join(testFolder, "copybook" + ".cpy");
+        const zoweApi = new ZoweApi();
+        const copybooksDownloadService: CopybookDownloadService = new CopybookDownloadService(undefined, zoweApi, undefined, undefined);
```
Comment (asatklichov) [inline clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts:181]: CopybookDownloadService sometimes created with null, sometimes with with undefined.  Even though they have differences, I think here it is used on  same purpose. Can use use one ?

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts [hunk 4]
```diff
@@ -12,119 +12,207 @@
...
+    function cleanupScenario() {
+        fs.remove(path.join(__dirname, C4Z_FOLDER));
+    }
+
+    test("With a valid configuration of dataset, copybook is created on FS", async () => {
+        const {copybookURIPath, copybooksDownloadService} = setupScenario();
+        await (copybooksDownloadService as any).downloadCopybookFromMFUsingZowe("dataset", "copybook", "profile");
+        expect(fs.existsSync(copybookURIPath)).toBe(true);
+
+        // remove folder and file created to run the scenario
```
Comment (asatklichov) [inline clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts:198]: we can remove these comments, method has good naming

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts (hunk 2 above)
Comment (zacanbrcom) [reply clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts:162]: NICE! I will do that way!

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts (hunk 1 above)
Comment (zacanbrcom) [reply clients/cobol-lsp-vscode-extension/src/__tests__/CopybookResolveURITest.ts:142]: OK

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts (hunk 3 above)
Comment (zacanbrcom) [reply clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts:181]: I will try to cleanup as much as possible

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts (hunk 4 above)
Comment (zacanbrcom) [reply clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts:198]: sure!

File: clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts [hunk 5]
```diff
@@ -12,119 +12,207 @@
...
-                    cbd.start();
-                } finally {
-                    cbd.dispose();
-                }
-
-                expect(zoweApi.listZOSMFProfiles).toBeCalled();
-        */
+vscode.workspace.workspaceFolders = [{} as any];
+vscode.window.showInformationMessage = () => Promise.resolve("Download Copybooks");
+const profile = "zoweProfile";
```
Comment (asatklichov) [inline clients/cobol-lsp-vscode-extension/src/__tests__/CopybookDownloadServiceTest.ts:29]: CopybookProfile("copybook", "zoweProfile") if const not reused

**Pull request merged**